File: src/lib/api/frontendRPCClient.ts
import * as pb from "../pb";
import { RPCHost } from "./rpc_host";
import { Readable as GenericReadable } from "./stream";

export default class FrontendRPC extends RPCHost {
  public createProfile(
    v: pb.ICreateProfileRequest = new pb.CreateProfileRequest()
  ): Promise<pb.CreateProfileResponse> {
    return this.expectOne(this.call("FrontendRPC/CreateProfile", new pb.CreateProfileRequest(v)));
  }
  public loadProfile(
    v: pb.ILoadProfileRequest = new pb.LoadProfileRequest()
  ): Promise<pb.LoadProfileResponse> {
    return this.expectOne(this.call("FrontendRPC/LoadProfile", new pb.LoadProfileRequest(v)));
  }
  public getProfile(
    v: pb.IGetProfileRequest = new pb.GetProfileRequest()
  ): Promise<pb.GetProfileResponse> {
    return this.expectOne(this.call("FrontendRPC/GetProfile", new pb.GetProfileRequest(v)));
  }
  public deleteProfile(
    v: pb.IDeleteProfileRequest = new pb.DeleteProfileRequest()
  ): Promise<pb.DeleteProfileResponse> {
    return this.expectOne(this.call("FrontendRPC/DeleteProfile", new pb.DeleteProfileRequest(v)));
  }
  public getProfiles(
    v: pb.IGetProfilesRequest = new pb.GetProfilesRequest()
  ): Promise<pb.GetProfilesResponse> {
    return this.expectOne(this.call("FrontendRPC/GetProfiles", new pb.GetProfilesRequest(v)));
  }
  public loadSession(
    v: pb.ILoadSessionRequest = new pb.LoadSessionRequest()
  ): Promise<pb.LoadSessionResponse> {
    return this.expectOne(this.call("FrontendRPC/LoadSession", new pb.LoadSessionRequest(v)));
  }
  public createNetwork(
    v: pb.ICreateNetworkRequest = new pb.CreateNetworkRequest()
  ): Promise<pb.CreateNetworkResponse> {
    return this.expectOne(this.call("FrontendRPC/CreateNetwork", new pb.CreateNetworkRequest(v)));
  }
  public deleteNetwork(
    v: pb.IDeleteNetworkRequest = new pb.DeleteNetworkRequest()
  ): Promise<pb.DeleteNetworkResponse> {
    return this.expectOne(this.call("FrontendRPC/DeleteNetwork", new pb.DeleteNetworkRequest(v)));
  }
  public getNetwork(
    v: pb.IGetNetworkRequest = new pb.GetNetworkRequest()
  ): Promise<pb.GetNetworkResponse> {
    return this.expectOne(this.call("FrontendRPC/GetNetwork", new pb.GetNetworkRequest(v)));
  }
  public getNetworks(
    v: pb.IGetNetworksRequest = new pb.GetNetworksRequest()
  ): Promise<pb.GetNetworksResponse> {
    return this.expectOne(this.call("FrontendRPC/GetNetworks", new pb.GetNetworksRequest(v)));
  }
  public getNetworkMemberships(
    v: pb.IGetNetworkMembershipsRequest = new pb.GetNetworkMembershipsRequest()
  ): Promise<pb.GetNetworkMembershipsResponse> {
    return this.expectOne(
      this.call("FrontendRPC/GetNetworkMemberships", new pb.GetNetworkMembershipsRequest(v))
    );
  }
  public deleteNetworkMembership(
    v: pb.IDeleteNetworkMembershipRequest = new pb.DeleteNetworkMembershipRequest()
  ): Promise<pb.DeleteNetworkMembershipResponse> {
    return this.expectOne(
      this.call("FrontendRPC/DeleteNetworkMembership", new pb.DeleteNetworkMembershipRequest(v))
    );
  }
  public createBootstrapClient(
    v: pb.ICreateBootstrapClientRequest = new pb.CreateBootstrapClientRequest()
  ): Promise<pb.CreateBootstrapClientResponse> {
    return this.expectOne(
      this.call("FrontendRPC/CreateBootstrapClient", new pb.CreateBootstrapClientRequest(v))
    );
  }
  public updateBootstrapClient(
    v: pb.IUpdateBootstrapClientRequest = new pb.UpdateBootstrapClientRequest()
  ): Promise<pb.UpdateBootstrapClientResponse> {
    return this.expectOne(
      this.call("FrontendRPC/UpdateBootstrapClient", new pb.UpdateBootstrapClientRequest(v))
    );
  }
  public deleteBootstrapClient(
    v: pb.IDeleteBootstrapClientRequest = new pb.DeleteBootstrapClientRequest()
  ): Promise<pb.DeleteBootstrapClientResponse> {
    return this.expectOne(
      this.call("FrontendRPC/DeleteBootstrapClient", new pb.DeleteBootstrapClientRequest(v))
    );
  }
  public getBootstrapClient(
    v: pb.IGetBootstrapClientRequest = new pb.GetBootstrapClientRequest()
  ): Promise<pb.GetBootstrapClientResponse> {
    return this.expectOne(
      this.call("FrontendRPC/GetBootstrapClient", new pb.GetBootstrapClientRequest(v))
    );
  }
  public getBootstrapClients(
    v: pb.IGetBootstrapClientsRequest = new pb.GetBootstrapClientsRequest()
  ): Promise<pb.GetBootstrapClientsResponse> {
    return this.expectOne(
      this.call("FrontendRPC/GetBootstrapClients", new pb.GetBootstrapClientsRequest(v))
    );
  }
  public createChatServer(
    v: pb.ICreateChatServerRequest = new pb.CreateChatServerRequest()
  ): Promise<pb.CreateChatServerResponse> {
    return this.expectOne(
      this.call("FrontendRPC/CreateChatServer", new pb.CreateChatServerRequest(v))
    );
  }
  public updateChatServer(
    v: pb.IUpdateChatServerRequest = new pb.UpdateChatServerRequest()
  ): Promise<pb.UpdateChatServerResponse> {
    return this.expectOne(
      this.call("FrontendRPC/UpdateChatServer", new pb.UpdateChatServerRequest(v))
    );
  }
  public deleteChatServer(
    v: pb.IDeleteChatServerRequest = new pb.DeleteChatServerRequest()
  ): Promise<pb.DeleteChatServerResponse> {
    return this.expectOne(
      this.call("FrontendRPC/DeleteChatServer", new pb.DeleteChatServerRequest(v))
    );
  }
  public getChatServer(
    v: pb.IGetChatServerRequest = new pb.GetChatServerRequest()
  ): Promise<pb.GetChatServerResponse> {
    return this.expectOne(this.call("FrontendRPC/GetChatServer", new pb.GetChatServerRequest(v)));
  }
  public getChatServers(
    v: pb.IGetChatServersRequest = new pb.GetChatServersRequest()
  ): Promise<pb.GetChatServersResponse> {
    return this.expectOne(this.call("FrontendRPC/GetChatServers", new pb.GetChatServersRequest(v)));
  }
  public startVPN(
    v: pb.IStartVPNRequest = new pb.StartVPNRequest()
  ): GenericReadable<pb.NetworkEvent> {
    return this.expectMany(this.call("FrontendRPC/StartVPN", new pb.StartVPNRequest(v)));
  }
  public stopVPN(v: pb.IStopVPNRequest = new pb.StopVPNRequest()): Promise<pb.StopVPNResponse> {
    return this.expectOne(this.call("FrontendRPC/StopVPN", new pb.StopVPNRequest(v)));
  }
  public joinSwarm(
    v: pb.IJoinSwarmRequest = new pb.JoinSwarmRequest()
  ): Promise<pb.JoinSwarmResponse> {
    return this.expectOne(this.call("FrontendRPC/JoinSwarm", new pb.JoinSwarmRequest(v)));
  }
  public leaveSwarm(
    v: pb.ILeaveSwarmRequest = new pb.LeaveSwarmRequest()
  ): Promise<pb.LeaveSwarmResponse> {
    return this.expectOne(this.call("FrontendRPC/LeaveSwarm", new pb.LeaveSwarmRequest(v)));
  }
  public startRTMPIngress(
    v: pb.IStartRTMPIngressRequest = new pb.StartRTMPIngressRequest()
  ): Promise<pb.StartRTMPIngressResponse> {
    return this.expectOne(
      this.call("FrontendRPC/StartRTMPIngress", new pb.StartRTMPIngressRequest(v))
    );
  }
  public startHLSEgress(
    v: pb.IStartHLSEgressRequest = new pb.StartHLSEgressRequest()
  ): Promise<pb.StartHLSEgressResponse> {
    return this.expectOne(this.call("FrontendRPC/StartHLSEgress", new pb.StartHLSEgressRequest(v)));
  }
  public stopHLSEgress(
    v: pb.IStopHLSEgressRequest = new pb.StopHLSEgressRequest()
  ): Promise<pb.StopHLSEgressResponse> {
    return this.expectOne(this.call("FrontendRPC/StopHLSEgress", new pb.StopHLSEgressRequest(v)));
  }
  public publishSwarm(
    v: pb.IPublishSwarmRequest = new pb.PublishSwarmRequest()
  ): Promise<pb.PublishSwarmResponse> {
    return this.expectOne(this.call("FrontendRPC/PublishSwarm", new pb.PublishSwarmRequest(v)));
  }
  public pProf(v: pb.IPProfRequest = new pb.PProfRequest()): Promise<pb.PProfResponse> {
    return this.expectOne(this.call("FrontendRPC/PProf", new pb.PProfRequest(v)));
  }
  public openChatServer(
    v: pb.IOpenChatServerRequest = new pb.OpenChatServerRequest()
  ): GenericReadable<pb.ChatServerEvent> {
    return this.expectMany(
      this.call("FrontendRPC/OpenChatServer", new pb.OpenChatServerRequest(v))
    );
  }
  public openChatClient(
    v: pb.IOpenChatClientRequest = new pb.OpenChatClientRequest()
  ): GenericReadable<pb.ChatClientEvent> {
    return this.expectMany(
      this.call("FrontendRPC/OpenChatClient", new pb.OpenChatClientRequest(v))
    );
  }
  public callChatClient(
    v: pb.ICallChatClientRequest = new pb.CallChatClientRequest()
  ): Promise<pb.CallChatClientResponse> {
    return this.expectOne(this.call("FrontendRPC/CallChatClient", new pb.CallChatClientRequest(v)));
  }
  public openVideoClient(
    v: pb.IVideoClientOpenRequest = new pb.VideoClientOpenRequest()
  ): GenericReadable<pb.VideoClientEvent> {
    return this.expectMany(
      this.call("FrontendRPC/OpenVideoClient", new pb.VideoClientOpenRequest(v))
    );
  }
  public openVideoServer(
    v: pb.IVideoServerOpenRequest = new pb.VideoServerOpenRequest()
  ): Promise<pb.VideoServerOpenResponse> {
    return this.expectOne(
      this.call("FrontendRPC/OpenVideoServer", new pb.VideoServerOpenRequest(v))
    );
  }
  public writeToVideoServer(
    v: pb.IVideoServerWriteRequest = new pb.VideoServerWriteRequest()
  ): Promise<pb.VideoServerWriteResponse> {
    return this.expectOne(
      this.call("FrontendRPC/WriteToVideoServer", new pb.VideoServerWriteRequest(v))
    );
  }
  public readMetrics(
    v: pb.IReadMetricsRequest = new pb.ReadMetricsRequest()
  ): Promise<pb.ReadMetricsResponse> {
    return this.expectOne(this.call("FrontendRPC/ReadMetrics", new pb.ReadMetricsRequest(v)));
  }
  public createNetworkInvitation(
    v: pb.ICreateNetworkInvitationRequest = new pb.CreateNetworkInvitationRequest()
  ): Promise<pb.CreateNetworkInvitationResponse> {
    return this.expectOne(
      this.call("FrontendRPC/CreateNetworkInvitation", new pb.CreateNetworkInvitationRequest(v))
    );
  }
  public createNetworkMembershipFromInvitation(
    v: pb.ICreateNetworkMembershipFromInvitationRequest = new pb.CreateNetworkMembershipFromInvitationRequest()
  ): Promise<pb.CreateNetworkMembershipFromInvitationResponse> {
    return this.expectOne(
      this.call(
        "FrontendRPC/CreateNetworkMembershipFromInvitation",
        new pb.CreateNetworkMembershipFromInvitationRequest(v)
      )
    );
  }
  public getBootstrapPeers(
    v: pb.IGetBootstrapPeersRequest = new pb.GetBootstrapPeersRequest()
  ): Promise<pb.GetBootstrapPeersResponse> {
    return this.expectOne(
      this.call("FrontendRPC/GetBootstrapPeers", new pb.GetBootstrapPeersRequest(v))
    );
  }
  public publishNetworkToBootstrapPeer(
    v: pb.IPublishNetworkToBootstrapPeerRequest = new pb.PublishNetworkToBootstrapPeerRequest()
  ): Promise<pb.PublishNetworkToBootstrapPeerResponse> {
    return this.expectOne(
      this.call(
        "FrontendRPC/PublishNetworkToBootstrapPeer",
        new pb.PublishNetworkToBootstrapPeerRequest(v)
      )
    );
  }
}

File: pkg/api/frontendrpc.client.go
package api

import (
	"context"

	"github.com/MemeLabs/go-ppspp/pkg/pb"
	"github.com/MemeLabs/go-ppspp/pkg/rpc"
)

type FrontendRPCClient struct {
	client *rpc.Client
}

// New ...
func NewFrontendRPCClient(client *rpc.Client) *FrontendRPCClient {
	return &FrontendRPCClient{client}
}

// CreateProfile ...
func (c *FrontendRPCClient) CreateProfile(
	ctx context.Context,
	req *pb.CreateProfileRequest,
	res *pb.CreateProfileResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/CreateProfile", req, res)
}

// LoadProfile ...
func (c *FrontendRPCClient) LoadProfile(
	ctx context.Context,
	req *pb.LoadProfileRequest,
	res *pb.LoadProfileResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/LoadProfile", req, res)
}

// GetProfile ...
func (c *FrontendRPCClient) GetProfile(
	ctx context.Context,
	req *pb.GetProfileRequest,
	res *pb.GetProfileResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/GetProfile", req, res)
}

// DeleteProfile ...
func (c *FrontendRPCClient) DeleteProfile(
	ctx context.Context,
	req *pb.DeleteProfileRequest,
	res *pb.DeleteProfileResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/DeleteProfile", req, res)
}

// GetProfiles ...
func (c *FrontendRPCClient) GetProfiles(
	ctx context.Context,
	req *pb.GetProfilesRequest,
	res *pb.GetProfilesResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/GetProfiles", req, res)
}

// LoadSession ...
func (c *FrontendRPCClient) LoadSession(
	ctx context.Context,
	req *pb.LoadSessionRequest,
	res *pb.LoadSessionResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/LoadSession", req, res)
}

// CreateNetwork ...
func (c *FrontendRPCClient) CreateNetwork(
	ctx context.Context,
	req *pb.CreateNetworkRequest,
	res *pb.CreateNetworkResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/CreateNetwork", req, res)
}

// DeleteNetwork ...
func (c *FrontendRPCClient) DeleteNetwork(
	ctx context.Context,
	req *pb.DeleteNetworkRequest,
	res *pb.DeleteNetworkResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/DeleteNetwork", req, res)
}

// GetNetwork ...
func (c *FrontendRPCClient) GetNetwork(
	ctx context.Context,
	req *pb.GetNetworkRequest,
	res *pb.GetNetworkResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/GetNetwork", req, res)
}

// GetNetworks ...
func (c *FrontendRPCClient) GetNetworks(
	ctx context.Context,
	req *pb.GetNetworksRequest,
	res *pb.GetNetworksResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/GetNetworks", req, res)
}

// GetNetworkMemberships ...
func (c *FrontendRPCClient) GetNetworkMemberships(
	ctx context.Context,
	req *pb.GetNetworkMembershipsRequest,
	res *pb.GetNetworkMembershipsResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/GetNetworkMemberships", req, res)
}

// DeleteNetworkMembership ...
func (c *FrontendRPCClient) DeleteNetworkMembership(
	ctx context.Context,
	req *pb.DeleteNetworkMembershipRequest,
	res *pb.DeleteNetworkMembershipResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/DeleteNetworkMembership", req, res)
}

// CreateBootstrapClient ...
func (c *FrontendRPCClient) CreateBootstrapClient(
	ctx context.Context,
	req *pb.CreateBootstrapClientRequest,
	res *pb.CreateBootstrapClientResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/CreateBootstrapClient", req, res)
}

// UpdateBootstrapClient ...
func (c *FrontendRPCClient) UpdateBootstrapClient(
	ctx context.Context,
	req *pb.UpdateBootstrapClientRequest,
	res *pb.UpdateBootstrapClientResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/UpdateBootstrapClient", req, res)
}

// DeleteBootstrapClient ...
func (c *FrontendRPCClient) DeleteBootstrapClient(
	ctx context.Context,
	req *pb.DeleteBootstrapClientRequest,
	res *pb.DeleteBootstrapClientResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/DeleteBootstrapClient", req, res)
}

// GetBootstrapClient ...
func (c *FrontendRPCClient) GetBootstrapClient(
	ctx context.Context,
	req *pb.GetBootstrapClientRequest,
	res *pb.GetBootstrapClientResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/GetBootstrapClient", req, res)
}

// GetBootstrapClients ...
func (c *FrontendRPCClient) GetBootstrapClients(
	ctx context.Context,
	req *pb.GetBootstrapClientsRequest,
	res *pb.GetBootstrapClientsResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/GetBootstrapClients", req, res)
}

// CreateChatServer ...
func (c *FrontendRPCClient) CreateChatServer(
	ctx context.Context,
	req *pb.CreateChatServerRequest,
	res *pb.CreateChatServerResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/CreateChatServer", req, res)
}

// UpdateChatServer ...
func (c *FrontendRPCClient) UpdateChatServer(
	ctx context.Context,
	req *pb.UpdateChatServerRequest,
	res *pb.UpdateChatServerResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/UpdateChatServer", req, res)
}

// DeleteChatServer ...
func (c *FrontendRPCClient) DeleteChatServer(
	ctx context.Context,
	req *pb.DeleteChatServerRequest,
	res *pb.DeleteChatServerResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/DeleteChatServer", req, res)
}

// GetChatServer ...
func (c *FrontendRPCClient) GetChatServer(
	ctx context.Context,
	req *pb.GetChatServerRequest,
	res *pb.GetChatServerResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/GetChatServer", req, res)
}

// GetChatServers ...
func (c *FrontendRPCClient) GetChatServers(
	ctx context.Context,
	req *pb.GetChatServersRequest,
	res *pb.GetChatServersResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/GetChatServers", req, res)
}

// StartVPN ...
func (c *FrontendRPCClient) StartVPN(
	ctx context.Context,
	req *pb.StartVPNRequest,
	res chan *pb.NetworkEvent,
) error {
	return c.client.CallStreaming(ctx, "FrontendRPC/StartVPN", req, res)
}

// StopVPN ...
func (c *FrontendRPCClient) StopVPN(
	ctx context.Context,
	req *pb.StopVPNRequest,
	res *pb.StopVPNResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/StopVPN", req, res)
}

// JoinSwarm ...
func (c *FrontendRPCClient) JoinSwarm(
	ctx context.Context,
	req *pb.JoinSwarmRequest,
	res *pb.JoinSwarmResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/JoinSwarm", req, res)
}

// LeaveSwarm ...
func (c *FrontendRPCClient) LeaveSwarm(
	ctx context.Context,
	req *pb.LeaveSwarmRequest,
	res *pb.LeaveSwarmResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/LeaveSwarm", req, res)
}

// StartRTMPIngress ...
func (c *FrontendRPCClient) StartRTMPIngress(
	ctx context.Context,
	req *pb.StartRTMPIngressRequest,
	res *pb.StartRTMPIngressResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/StartRTMPIngress", req, res)
}

// StartHLSEgress ...
func (c *FrontendRPCClient) StartHLSEgress(
	ctx context.Context,
	req *pb.StartHLSEgressRequest,
	res *pb.StartHLSEgressResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/StartHLSEgress", req, res)
}

// StopHLSEgress ...
func (c *FrontendRPCClient) StopHLSEgress(
	ctx context.Context,
	req *pb.StopHLSEgressRequest,
	res *pb.StopHLSEgressResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/StopHLSEgress", req, res)
}

// PublishSwarm ...
func (c *FrontendRPCClient) PublishSwarm(
	ctx context.Context,
	req *pb.PublishSwarmRequest,
	res *pb.PublishSwarmResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/PublishSwarm", req, res)
}

// PProf ...
func (c *FrontendRPCClient) PProf(
	ctx context.Context,
	req *pb.PProfRequest,
	res *pb.PProfResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/PProf", req, res)
}

// OpenChatServer ...
func (c *FrontendRPCClient) OpenChatServer(
	ctx context.Context,
	req *pb.OpenChatServerRequest,
	res chan *pb.ChatServerEvent,
) error {
	return c.client.CallStreaming(ctx, "FrontendRPC/OpenChatServer", req, res)
}

// OpenChatClient ...
func (c *FrontendRPCClient) OpenChatClient(
	ctx context.Context,
	req *pb.OpenChatClientRequest,
	res chan *pb.ChatClientEvent,
) error {
	return c.client.CallStreaming(ctx, "FrontendRPC/OpenChatClient", req, res)
}

// CallChatClient ...
func (c *FrontendRPCClient) CallChatClient(
	ctx context.Context,
	req *pb.CallChatClientRequest,
	res *pb.CallChatClientResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/CallChatClient", req, res)
}

// OpenVideoClient ...
func (c *FrontendRPCClient) OpenVideoClient(
	ctx context.Context,
	req *pb.VideoClientOpenRequest,
	res chan *pb.VideoClientEvent,
) error {
	return c.client.CallStreaming(ctx, "FrontendRPC/OpenVideoClient", req, res)
}

// OpenVideoServer ...
func (c *FrontendRPCClient) OpenVideoServer(
	ctx context.Context,
	req *pb.VideoServerOpenRequest,
	res *pb.VideoServerOpenResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/OpenVideoServer", req, res)
}

// WriteToVideoServer ...
func (c *FrontendRPCClient) WriteToVideoServer(
	ctx context.Context,
	req *pb.VideoServerWriteRequest,
	res *pb.VideoServerWriteResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/WriteToVideoServer", req, res)
}

// ReadMetrics ...
func (c *FrontendRPCClient) ReadMetrics(
	ctx context.Context,
	req *pb.ReadMetricsRequest,
	res *pb.ReadMetricsResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/ReadMetrics", req, res)
}

// CreateNetworkInvitation ...
func (c *FrontendRPCClient) CreateNetworkInvitation(
	ctx context.Context,
	req *pb.CreateNetworkInvitationRequest,
	res *pb.CreateNetworkInvitationResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/CreateNetworkInvitation", req, res)
}

// CreateNetworkMembershipFromInvitation ...
func (c *FrontendRPCClient) CreateNetworkMembershipFromInvitation(
	ctx context.Context,
	req *pb.CreateNetworkMembershipFromInvitationRequest,
	res *pb.CreateNetworkMembershipFromInvitationResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/CreateNetworkMembershipFromInvitation", req, res)
}

// GetBootstrapPeers ...
func (c *FrontendRPCClient) GetBootstrapPeers(
	ctx context.Context,
	req *pb.GetBootstrapPeersRequest,
	res *pb.GetBootstrapPeersResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/GetBootstrapPeers", req, res)
}

// PublishNetworkToBootstrapPeer ...
func (c *FrontendRPCClient) PublishNetworkToBootstrapPeer(
	ctx context.Context,
	req *pb.PublishNetworkToBootstrapPeerRequest,
	res *pb.PublishNetworkToBootstrapPeerResponse,
) error {
	return c.client.CallUnary(ctx, "FrontendRPC/PublishNetworkToBootstrapPeer", req, res)
}
